### DISCUSSION TITLE
Make recipe keywords an array

### DISCUSSION THREAD

**Discussion by christianlupus** [nextcloud/cookbook#1085] [Ideas and discussions]
In a JSON+LD object, the `keywords` are an array of entries. We are currently mapping that into a comma-separated list of keywords stored in the `*.json` files. On the frontend, the array is reconstructed and used in a `v-for` Vue attribute.

Example locations:
- https://github.com/nextcloud/cookbook/blob/f299ecc6d48f109bfe4214e8fac5877d97afadfc/lib/Service/RecipeService.php#L292-L294
- https://github.com/nextcloud/cookbook/blob/f299ecc6d48f109bfe4214e8fac5877d97afadfc/src/components/RecipeEdit.vue#L518-L523

I am considering this a bad idea to do such mapping back and forth without any obvious benefit. I do not see such a benefit at the moment. However, I might be misled and there _is_ in fact a reason for this. I am asking for you opinions on this. Any comments, @seyfeb?

**Comment by seyfeb** [comment]
I don't know of and don't see a reason. I implemented this in the frontend like this only because that was the data the API returned!

**Reply by christianlupus** [reply]
All right, no offense. I just wanted to make sure, i am not dumb. 

I will create a wishlist for that in the issues then.

**Reply by seyfeb** [reply]
None taken ;)
